### ISSUE TITLE
Potential bug in cdbpath_partkeys_from_preds()

### ISSUE THREAD

**Issue by pivotal-ning-yu** [greenplum-db/gpdb#5370]
In function `cdbpath_partkeys_from_preds()` there is below logic:

https://github.com/greenplum-db/gpdb/blob/04e39f4a3c116e323c0fc0f47e0c6b8cd790537f/src/backend/cdb/cdbpath.c#L696-L704

In L703 we are appending a `EquivalenceClass *` item to a `PathKey *` list, this is obviously wrong. Maybe we should change it like below:

```C
 	if (pathkey->pk_eclass == rinfo->left_ec)
 	{
 		a_partkey = lappend(a_partkey, makePathKeyForEC(rinfo->left_ec));
		break;
	}
```

If we do not break immediately we'll run into endless loop.

**Comment by zhangjackey** [comment]
The legacy planner in 5.8 and 5.9 will always broadcast the table even very big table for the following case, while ORCA can generate expected redistribution motion plan.

reproduce steps:
```
set optimizer=off;
drop table if exists hello;
drop table if exists hello2;
create table hello(id int,name text) distributed by(id);
insert into hello select i,i from  generate_series(1,1000) i;
create table hello2 as select * from hello distributed by (id);
analyze hello;
analyze hello2;
explain select * from hello a ,hello2 b where a.name=b.name;
```

still broadcast when set gp_segments_for_planner =10000; 

the same for 100M rows big table.

**Comment by pivotal-ning-yu** [comment]
Fixed by #5539
